### ISSUE TITLE
Build Process - Deprecation notifications

### ISSUE THREAD

**Issue by clepore** [infor-design/design-system#309]
We need to create a way to deprecate and notify users. This will also need to be a system that can be used on all repo's, but we will get it functional here first, before spreading it around.

Requirements (specifically for this repo):
- `./deprecated-log.json` file with a list of all deprecations
- a way to log any looming deprecation notices to a user when they do an `npm install`
- a way to add any pending deprecations to github release notes

**Comment by clepore** [comment]
Some thoughts here: https://css-tricks.com/approaches-to-deprecating-code-in-javascript/

**Comment by nickwynja** [comment]
I think the `messages.json`/`deprecated.json` should have some of this information:

- `versionDeprecated`: the version that the deprecation was added
- `versionToBeRemoved`: the version that the deprecated feature will be removed
- `message`: Text describing how to handle the deprecation, or other warning if not a deprecation
- `messageLevel`: a level for the message, like `info`, `warn`, `critical`

I think the only item of the object that should be required is `message` since we could have an info notice like: 

```
Notice: 'Uplift' and 'Future' assets are considered alpha and not yet fully supported.
```

**Comment by clepore** [comment]
I agree with all of that, but I think instead of 3 levels, we should do `isCritical: true|false`.

**Comment by nickwynja** [comment]
Closed by #319 #320
